src/peer.rs

```diff
@@ -173,10 +188,7 @@ pub(crate) fn new_peer<T: HandshakeHandler>(
                 Ok(data) => {
                     if data.is_empty() {
                         println!("Peer stop");
-                        {
-                            let mut active_connections = active_connections.write();
-                            active_connections.connections.remove(&peer_id);
-                        }
+                        active_connections.write().connections.remove(&peer_id).expect("Unable to remove peer id");
```

Comment:
expect() is intentional ?

Reply:
Yes this should never happens but maybe a print could be enough as even if it happens it will not critically break the code.